src/main/java/school/sorokin/javacore/spring_core_homework_base/Services/AccountService.java

```
@@ -1,111 +1,104 @@
+        Account fromAccount = session.find(Account.class, from);
+        Account toAccount = session.find(Account.class, to);
 
+        if (fromAccount == null) {
+            throw new TransferAccountException("Не найден аккаунт отправителя");
+        }
+        if (toAccount == null) {
+            throw new TransferAccountException("Не найден аккаунт получателя");
+        }
+        if (fromAccount.getMoneyAmount().compareTo(sum) < 0) {
+            throw new TransferAccountException("Недостаточно денег для перевода со счета на счёт");
+        }
+        if (fromAccount.getUser().getId().equals(toAccount.getUser().getId())) {
+            fromAccount.setMoneyAmount(fromAccount.getMoneyAmount().subtract(sum));
+            toAccount.setMoneyAmount(toAccount.getMoneyAmount().add(sum));
+            System.out.println("Транзакция между своими счетами прошла успешно, комиссии нет");
         } else {
             BigDecimal percent = sum.multiply(commissionPayment).divide(BigDecimal.valueOf(100));
-            if (accountFrom.getMoneyAmount().compareTo(sum.add(percent)) >= 0) {
-                accountFrom.setMoneyAmount(accountFrom.getMoneyAmount().subtract(sum.add(percent)));
-                accountTo.setMoneyAmount(accountTo.getMoneyAmount().add(sum));
+            if (fromAccount.getMoneyAmount().compareTo(sum.add(percent)) >= 0) {
+                fromAccount.setMoneyAmount(fromAccount.getMoneyAmount().subtract(sum.add(percent)));
+                toAccount.setMoneyAmount(toAccount.getMoneyAmount().add(sum));
                 System.out.println("Транзакция между чужими счетами прошла успешно, комиссия: " + percent);
-            }else {
-                throw new TransferFromAccounException("Недостаточно средств для перевода на чужой счёт(комиссия 1,5%)");
+            } else {
+                throw new TransferAccountException("Недостаточно средств для перевода на чужой счёт(комиссия 1,5%)");
```

также сначала кидаем исключение, потом уже основная логика fail fast